cluster-operator/src/main/java/io/strimzi/operator/cluster/model/metrics/StrimziMetricsReporterModel.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.operator.cluster.model.metrics;
+
+import io.strimzi.api.kafka.model.common.HasConfigurableMetrics;
+import io.strimzi.api.kafka.model.common.metrics.StrimziReporterMetrics;
+import io.strimzi.kafka.oauth.common.ConfigException;
+import io.strimzi.operator.common.model.InvalidResourceException;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.regex.Pattern;
+import java.util.regex.PatternSyntaxException;
+
+/**
+ * Represents a model for components with configurable metrics using Strimzi Reporter
+ */
+public class StrimziMetricsReporterModel {
+
+    /**
+     * Name of the Strimzi metrics port
+     */
+    public static final String METRICS_PORT_NAME = "http-prometheus";
+
+    /**
+     * Number of the Strimzi metrics port
+     */
+    public static final int METRICS_PORT = 9404;
+    private final boolean isEnabled;
+    private final List<String> allowList;
+
+    /**
+     * Constructs the StrimziMetricsReporterModel for managing configurable metrics with Strimzi Reporter
+     *
+     * @param specSection StrimziReporterMetrics object containing the metrics configuration
+     */
+    public StrimziMetricsReporterModel(HasConfigurableMetrics specSection) {
+        if (specSection.getMetricsConfig() != null) {
+            if (specSection.getMetricsConfig() instanceof StrimziReporterMetrics spec) {
+                validate(spec);
+                this.isEnabled = true;
+                this.allowList = spec.getValues() != null &&
+                        spec.getValues().getAllowList() != null
+                        ? spec.getValues().getAllowList() : null;
+            } else {
+                throw new ConfigException("Unsupported metrics type " + specSection.getMetricsConfig().getType());
```

Comment:
Looking at MetricsModel, this should be `InvalidResourceException`.

Reply:
Refactored